tests/phpunit/tests/rest-api/rest-post-meta-fields.php

```diff
@@ -3112,4 +3112,93 @@ public function error_delete_query( $query ) {
 		}
 		return $query;
 	}
+
+	/**
+	 * Test that post meta is revisioned when saving to the REST API as part of a post update.
+	 *
+	 * @ticket 20564
+	 */
+	public function test_revisioned_post_meta() {
+		$this->grant_write_permission();
+
+		register_post_meta(
+			'post',
+			'foo',
+			array(
+				'single'       => true,
+				'show_in_rest' => true,
+			)
+		);
+
+		$post_id = self::$post_id;
+
+		// Create a revision.
+		$request = new WP_REST_Request( 'POST', sprintf( '/wp/v2/posts/%d/revisions', $post_id ) );
+		$request->set_body_params(
+			array(
+				'title' => 'Initial revision',
+				'meta'  => array(
+					'foo' => 'initial',
+				),
+			)
+		);
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertSame( 201, $response->get_status() );
+		$revision_id = $response->get_data()['id'];
+
+		// Update the post.
+		$request = new WP_REST_Request( 'PUT', sprintf( '/wp/v2/posts/%d', $post_id ) );
+		$request->set_body_params(
+			array(
+				'title' => 'Revision 1',
+				'meta'  => array(
+					'foo' => 'bar',
+				),
+			)
+		);
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertSame( 200, $response->get_status() );
+
+		// Check that the revision has the old value.
+		$request  = new WP_REST_Request( 'GET', sprintf( '/wp/v2/posts/%d/revisions/%d', $post_id, $revision_id ) );
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertSame( 200, $response->get_status() );
+		$this->assertSame( 'bar', $response->get_data()['meta']['foo'] );
+
+		// Create two more revisions with different meta values for the foo key.
+		$request = new WP_REST_Request( 'POST', sprintf( '/wp/v2/posts/%d/revisions', $post_id ) );
```

Comment:
It would be good to test creating revisions through the `/wp/v2/posts/%d` endpoint instead of manually creating revisions.

Reply:
added